interp/interp.go

```diff
@@ -80,6 +85,7 @@ type Interpreter struct {
 	binPkg   Exports           // binary packages used in interpreter, indexed by path
 	srcPkg   imports           // source packages used in interpreter, indexed by path
 	rdir     map[string]bool   // for src import cycle detection
+	runid                      // for cancellation
```

Comment:
This needs to be protected from races. This will add synchronisation overhead, but will be less than for the select approach.